src/Domain/Campaign.php

```diff
@@ -409,6 +429,10 @@ final public function updateFromSfPull(
         Assertion::nullOrRegex($status, "/^[A-Za-z]{2,30}$/");
         Assertion::betweenLength($name, 2, 255);
         Assertion::nullOrMaxLength($thankYouMessage, 500);
+        Assertion::nullOrBetweenLength($metaCampaignSlug, 1, 64);
+        Assertion::nullOrRegex($metaCampaignSlug, '/^[-a-z50-9]+$/');
```

Comment:
```suggestion
        Assertion::nullOrRegex($metaCampaignSlug, '/^[-a-z0-9]+$/');
```
I think in this case the `5` is harmless, just a little confusing.

I wonder if we should allow uppercase A-Z – I think it may be the only range that is [allowed on the SF field](https://github.com/thebiggive/salesforce/blob/cb006cb0251ffa8856cdea172157029e4422b9a3/force-app/main/default/objects/ChampionFunding__c/validationRules/Slug_Uses_Valid_Characters.validationRule-meta.xml#L7-L8) but not here, but in practice I can't remember seeing an uppercase character used.

Reply:
Thanks, 5 removed and uppercase allowed in https://github.com/thebiggive/matchbot/pull/1416/commits/4cf2b4f471176792abc5f179dfa3c5b49e1a66b7

I actually discovered after writing this regex that I'd already created a MetaCampaignSlug VO class and forgoton about it - that has a regex `^[A-Za-z0-9-]+$` so changed to the same thing here, although IMHO it's better that we never use capitals in a slug just in case they ever get matched in a case sensitive way.

Should really DRY up and re-use the regex and slightly different max length from the VO, but not worth changing right now to me. Not using the VO as a property type since embeddables are a bit too awkward with doctrine.